core/presentation/designsystem/src/main/java/com/pronoidsoftware/core/presentation/designsystem/components/TaskyActionButton.kt

```diff
@@ -0,0 +1,84 @@
+package com.pronoidsoftware.core.presentation.designsystem.components
+
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.IntrinsicSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.layout.widthIn
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material3.Button
+import androidx.compose.material3.ButtonDefaults
+import androidx.compose.material3.CircularProgressIndicator
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.alpha
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import com.pronoidsoftware.core.presentation.designsystem.LocalSpacing
+import com.pronoidsoftware.core.presentation.designsystem.TaskyDarkGray
+import com.pronoidsoftware.core.presentation.designsystem.TaskyLightGray
+import com.pronoidsoftware.core.presentation.designsystem.TaskyTheme
+
+@Composable
+fun TaskyActionButton(
+    text: String,
+    isLoading: Boolean,
+    onClick: () -> Unit,
+    modifier: Modifier = Modifier,
+    enabled: Boolean = true,
+) {
+    val spacing = LocalSpacing.current
+    Button(
+        onClick = onClick,
+        enabled = enabled,
+        colors = ButtonDefaults.buttonColors(
+            containerColor = MaterialTheme.colorScheme.primaryContainer,
+            contentColor = MaterialTheme.colorScheme.onPrimaryContainer,
+            disabledContainerColor = TaskyLightGray,
+            disabledContentColor = TaskyDarkGray,
+        ),
+        shape = RoundedCornerShape(100f),
+        modifier = modifier
+            .widthIn(max = spacing.maxButtonWidth)
+            .height(IntrinsicSize.Min),
+    ) {
+        Box(
+            modifier = Modifier
+                .fillMaxWidth()
+                .padding(vertical = spacing.spaceMediumSmall),
+            contentAlignment = Alignment.Center,
+        ) {
+            CircularProgressIndicator(
+                modifier = Modifier
+                    .size(15.dp)
+                    .alpha(if (isLoading) 1f else 0f),
+                strokeWidth = spacing.strokeBold,
+                color = MaterialTheme.colorScheme.onPrimaryContainer,
+            )
+            Text(
+                text = text,
+                modifier = Modifier
+                    .alpha(if (isLoading) 0f else 1f),
+                style = MaterialTheme.typography.labelLarge,
+            )
+        }
```

Comment:
This structure is also very similar as for Runique, but someone actually commented a suggestion there that the box isn't needed, since the loading indicator and text will never be shown together

Reply:
Yes, you should be able to center it with the RowScope